content/exchange/artifacts/Windows.EventLogs.Hayabusa.yaml

```diff
@@ -60,18 +60,18 @@ parameters:
 sources:
  - query: |
 
-        LET Toolzip <= SELECT FullPath FROM Artifact.Generic.Utils.FetchBinary(ToolName="Hayabusa-2.2", IsExecutable=FALSE)
+        LET Toolzip <= SELECT FullPath FROM Artifact.Generic.Utils.FetchBinary(ToolName="Hayabusa", IsExecutable=FALSE)
```

Comment:
need to refer to the same tool you define above.

Reply:
@scudette I see, I noticed that you changed the tool name above from Hayabusa to Hayabusa-2.2 in my previous PR but I was not sure why. Should we always specify the tool version in the artifact name?

Reply:
The reason is that in older versions of Velociraptor it was hard to upgrade tool definitions. So users who already had downloaded the older versions could not upgrade the tool to the new version. The intention was that the version of the tool goes with the tool name. So if we have multiple artifacts using the same tool they can operate independently. 

So for example say artifact A declares a tool X then artifact B declares the same tool. But then artifact A upgrades to a new version of the tool, will this break artifact B? It's not a problem if the version of the tool goes with the name. Especially if the versions are not compatible.

If your tool is hosted on GitHub it's a lot easier to just specify the GitHub release and project name because then Velociraptor can handle upgrading the tool without changing the tool definition.

In 0.6.8 there is a UI that people can change the version of the tool installed so it's not as problematic but in older versions it's a bit annoying.

Reply:
To specify the tool as a GitHub project 
https://github.com/Velocidex/velociraptor/blob/b628927a76977fac32ab112e4c95bee7eece9119/artifacts/definitions/Server/Internal/ToolDependencies.yaml#L8

```
  - name: VelociraptorWindows
    github_project: Velocidex/velociraptor
    github_asset_regex: windows-amd64.exe
    serve_locally: true
```

The github_asset_regex is a reg that will be used to select the first tool that matches under the latest project release. This ensures that users can upgrade automatically without having to change the artifact at all. Of course it also means that the tool guarantees at least command line compatibility or the artifact will break so it depends on how stable the tool is to whether this is a good approach.

Reply:
@scudette I see. Thank you for the explanation. The automatic upgrade is a great feature but I cannot guarantee the command line syntax will always be the same so for now I would like to keep it as it is. I renamed the tool name to just 'Hayabusa'

Reply:
You should name the tool according to the version especially if you can not guarantee future compatibility. See https://semver.org/ for more information about versioning.

The issue is that those users who had previous had just "Hayabusa" installed will be unable to upgrade if we do not attach the version to the tool name. This is why I added the version to the tool name in the first place.

Reply:
To clarify this is the situation:

A user had previously installed this artifact from the exchange and the tool "Hayabusa" was defined with 2.0.0 version. The inventory downloaded that version and now each time an artifact wants to use "Hayabusa" they get served that tool.

Now the user installs the new version of this artifact (e.g. from the exchange). This artifact still calls the tool "Hayabusa" which means Velociraptor already has the tool in the inventory so it happily just serves what it has. So we get the new artifact running with the old tool - user can not upgrade to Hayabusa 2.2.

After 0.6.8 this will still be the default behavior but there is now a button that tells Velociraptor- reset this definition anyway and start again. Which from a usability perspective is still not great (user has to do something for proper work) but at least they can do something about it.

If you use Hayabusa-2.2-2 as the tool name everything just works with the artifact downloading the correct version it is designed to work with. Any other way will cause potential for breakage.